service/availability_check.go

```diff
@@ -72,8 +72,14 @@ func RequestAvailabilityCheck(c echo.Context, source *m.Source, headers []kafka.
 
 // sends off an availability check http request for each of the source's
 // applications
-func (acr availabilityCheckRequester) ApplicationAvailabilityCheck(source *m.Source) {
+func (acr availabilityCheckRequester) ApplicationAvailabilityCheck(source *m.Source, skipEmptySources bool) {
 	for _, app := range source.Applications {
+		// We do not want to send availability check requests for Cost Management applications, as detailed in
+		// https://issues.redhat.com/browse/RHCLOUD-38735.
+		if skipEmptySources && app.ApplicationType.Name == "/insights/platform/cost-management" {
```

Comment:
Nit-picky...what about moving the `app.ApplicationType.Name == "/insights/platform/cost-management"` logic into a method?  Something like the following `isCostManagementApp(app)`?

```
func isCostManagementApp(app) bool {
  return app.ApplicationType.Name == "/insights/platform/cost-management"
}
```

Would that help communicate what that bit of logic is actually doing (kind of self-commenting) and hide the details about how it does (in case the logic changes in the future)?